src/modules/job-list/stats.c

```diff
@@ -103,9 +151,14 @@ static void stats_purge (struct job_stats *stats, struct job *job)
  */
 void job_stats_purge (struct job_stats_ctx *statsctx, struct job *job)
 {
+    struct job_stats *stats;
+
     assert (job->state == FLUX_JOB_STATE_INACTIVE);
```

Comment:
An assertion here kills the rank 0 broker.  Appropriate in this case?

Reply:
i think that assert was meant as a double check that this is only called on inactive jobs, but I see

```
    if (flux_event_unpack (msg, NULL, "{s:o}", "jobs", &jobs) < 0)                                                                          
        flux_log_error (h, "job-purge-inactive message");                                                                                   
    json_array_foreach (jobs, index, entry) {                                                                                               
        flux_jobid_t id = json_integer_value (entry);                                                                                       
        struct job *job;                                                                                                                    
                                                                                                                                            
        if ((job = zhashx_lookup (ctx->jsctx->index, &id))) {                                                                               
            job_stats_purge (ctx->jsctx->statsctx, job);                                                                                    
            if (job->list_handle)                                                                                                           
                zlistx_delete (ctx->jsctx->inactive, job->list_handle);                                                                     
            zhashx_delete (ctx->jsctx->index, &id);                                                                                         
            count++;                                                                                                                        
        }                                                                                                                                   
    }                                                          
```

that we do not verify this fact.  it trusts the fact that every job id passed to it is inactive.  We should double check that within the loop.

I'll create an issue, b/c this is independent of this PR.